javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/model/typesystem/LazyType.java

```diff
@@ -80,20 +52,32 @@ public LazyType(Supplier<ResolvedType> supplier) {
         this.supplier = supplier;
     }
 
-    private ResolvedType getType() {
+    /**
+     * Get the type wrapped by a {@link LazyType}.
+     *
+     * This method is used to get and unpack the {@link LazyType} when they are nested.
+     *
+     * {@code
+     *      LazyType typeA = new LazyType(() -> ResolvedVoidType.INSTANCE);
+     *      LazyType typeB = new LazyType(() -> typeA);
+     *      LazyType typeC = new LazyType(() -> typeB);
+     * }
+     *
+     * In this examples, by calling {@code getType()} we would expected all of them to return {@link ResolvedVoidType}
+     *
+     * @return The concrete type for the {@link LazyType}.
+     */
+    public ResolvedType getType() {
         if (concrete == null) {
             concrete = supplier.get();
         }
-        return concrete;
-    }
 
-    /**
-     * Get the concrete {@link ResolvedType} for the current {@link LazyType}.
-     *
-     * @return The {@link ResolvedType} if already resolved, otherwise empty.
-     */
-    public Optional<ResolvedType> getConcreteType() {
-        return Optional.ofNullable(concrete);
+        // If the concrete value is a lazy type, then get the inner type.
+        if (concrete instanceof LazyType) {
```

Comment:
Is it possible or autorized to build a LazyType(LazyType(Type))? What could be the meaning of this?

Reply:
> What could be the meaning of this?

To be honest i don't see any point in doing this. But we don't know the type of the wrapped type until we ask for it.
So i think it's better to play safe and assume the user may do this.

Reply:
If there is no identified use case, perhaps it is sufficient to protect the constructor against a LazyType nesting.

Reply:
The problem is we don't provide Type in the constructor, we provide a `Supplier` that will create only if needed.

If we try to check the type of it in the constructor we are breaking the lazy principle (object is only created when needed).